components/integrations/DisconnectIntegration.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from "react";
+import { useMutation } from "react-query";
+
+import { Dialog } from "@components/Dialog";
+import ConfirmationDialogContent from "@components/dialog/ConfirmationDialogContent";
+import { ButtonBaseProps } from "@components/ui/Button";
+
+export default function DisconnectIntegration(props: {
+  /** Integration credential id */
+  id: number;
+  render: (renderProps: ButtonBaseProps) => JSX.Element;
+  onOpenChange: (isOpen: boolean) => void | Promise<void>;
+}) {
+  const [modalOpen, setModalOpen] = useState(false);
+  const mutation = useMutation(
+    async () => {
+      const res = await fetch("/api/integrations", {
+        method: "DELETE",
+        body: JSON.stringify({ id: props.id }),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      if (!res.ok) {
+        throw new Error("Something went wrong");
+      }
+    },
+    {
+      async onSettled() {
+        props.onOpenChange(modalOpen);
```

Comment:
makes sure the mutation doesn't leave loading until the promise is resolved
```suggestion
        await props.onOpenChange(modalOpen);
```

Reply:
Sorry @KATT automerge was enabled. Will make a quick PR